packages/extension-ui/src/components/ValidatedInput.tsx

```diff
@@ -26,6 +26,12 @@ function ValidatedInput<T extends Record<string, unknown>> ({ className, compone
   const [validationResult, setValidationResult] = useState<Result<string>>(Result.ok(''));
   const isMounted = useIsMounted();
 
+  useEffect(() => {
```

Comment:
To me, this `useEffect` shouldn't be needed.. because the `defaultValue` prop change should make the whole `ValidatedInput` component rerender. Yet, it doesn't.. and the input isn't populated correctly without this `useEffect`. Does this come from the props type extending a `Record<string, string>`?